src/main/java/net/mcreator/ui/modgui/ItemExtensionGUI.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2022, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the tefms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.modgui;
+
+import net.mcreator.blockly.data.Dependency;
+import net.mcreator.element.types.ItemExtension;
+import net.mcreator.minecraft.ElementUtil;
+import net.mcreator.ui.MCreator;
+import net.mcreator.ui.MCreatorApplication;
+import net.mcreator.ui.component.JEmptyBox;
+import net.mcreator.ui.component.util.PanelUtils;
+import net.mcreator.ui.help.HelpUtils;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.ui.minecraft.MCItemHolder;
+import net.mcreator.ui.procedure.ProcedureSelector;
+import net.mcreator.ui.validation.AggregatedValidationResult;
+import net.mcreator.workspace.elements.ModElement;
+import net.mcreator.workspace.elements.VariableTypeLoader;
+
+import javax.annotation.Nullable;
+import javax.swing.*;
+import javax.swing.border.TitledBorder;
+import java.awt.*;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+public class ItemExtensionGUI extends ModElementGUI<ItemExtension> {
+	private final MCItemHolder item = new MCItemHolder(mcreator, ElementUtil::loadBlocksAndItems);
+
+	// Fuel
+	private final JCheckBox enableFuel = L10N.checkbox("elementgui.common.enable");
+	private final JSpinner fuelPower = new JSpinner(new SpinnerNumberModel(1600, 0, Integer.MAX_VALUE, 1));
+	// Compostable
+	private final JSpinner layerChance = new JSpinner(new SpinnerNumberModel(0, 0.00f, 1.00f, 0.05f));

```

Comment:
I'd replace the spinner model to `(0, 0, 1, 0.01)` so that it uses doubles and allows more precise values

Reply:
Users should not use a double value because Minecraft uses a float variable for the chance. I am only using a double to save it because the number spinner object uses a double.

Reply:
I tested with the current values and the spinner doesn't work properly (if it's at 0.05, it wouldn't go down to 0, same for 0.95 to 1). Moving to double did fix the issue